superset/sql_parse.py

```diff
@@ -378,3 +380,23 @@ def set_or_update_query_limit(self, new_limit: int, force: bool = False) -> str:
         for i in statement.tokens:
             str_res += str(i.value)
         return str_res
+
+
+def validate_filter_clause(clause: str) -> None:
```

Comment:
I agree this module is slightly messy right now, but it's not totally clear to me how this functionality will evolve. Since we already have some util-type of functionality, I will add this here for now, but going forward this should probably be refactored. But that refactoring should not be done as part of this PR.